### PR TITLE
Improve message for at-test_throw fails

### PR DESCRIPTION
**Pull request by IainNZ**
Partially (and maybe completely) addresses https://github.com/JuliaLang/julia/issues/14239

### CODE REVIEW

File: base/test.jl [hunk 1]
```diff
@@ -72,9 +72,13 @@ function Base.show(io::IO, t::Fail)
     print_with_color(:red, io, "Test Failed\n")
     print(io, "  Expression: ", t.orig_expr)
     if t.test_type == :test_throws
-        # Either no exception, or wrong exception
         print(io, "\n    Expected: ", t.expr)
-        print(io, "\n      Thrown: ", typeof(t.value))
+        # Either no exception, or wrong exception
+        if t.value == nothing
```
Comment (tkelman) [inline base/test.jl]: I think `===` is preferred for comparison to `nothing`

**Comment by JeffBezanson** [comment]
I really think we should accurately keep track of whether an exception was actually thrown, instead of relying on a sentinel value.

**Comment by IainNZ** [comment]
Thats fair, I can do that

**Comment by IainNZ** [comment]
Updated PR

**Comment by tkelman** [comment]
should the message here be metatested?

**Comment by IainNZ** [comment]
Had to think a bit about how to do that @tkelman, but I've put that in now too. Will merge when green.

**Comment by tkelman** [comment]
+1, who tests the testers? @IainNZ, that's who.

**Pull request merged**